apps/backend/src/app/modules/maps/maps.service.ts

```diff
@@ -1843,6 +1883,27 @@ export class MapsService {
       throw new BadRequestException('BSP is not compressed');
   }
 
+  async getBspFromTemp(userID: number): Promise<File> {
+    const userIDObjects = await this.fileStoreService.listFileKeys(
+      `upload_tmp/${userID}`
+    );
+
+    if (userIDObjects.length === 0) {
+      throw new BadRequestException('Missing BSP file');
+    }
+
+    const headerData = await this.fileStoreService.getFile(userIDObjects[0]);
```

Comment:
Weird variable name - isn't this the whole buffer?

Reply:
It was just a header before

Reply:
It's a byte array!